pkg/crc/preflight/preflight_checks_nonlinux.go

```diff
@@ -70,3 +73,13 @@ func olderDaemonVersionRunning() error {
 	}
 	return daemonclient.CheckIfOlderVersion(v)
 }
+
+func checkIfDaemonIsServing() error {
+	return crcerrors.Retry(context.Background(), 15*time.Second, func() error {
```

Comment:
15 seconds is a very long time, if it's not ready to serve requests after a few seconds (3?), I'm not sure it's worth waiting 15 seconds.
Does this have to be done in a separate preflight? Or could this wait be added right after the daemon start, with a big warning if it fails?

Reply:
3 sec is not enough, I checked on my machine in the loop and looks like 15 sec is best bet sometime it starts in 7-8 secs but sometime it takes around 10-11 secs when I have lot of I/O happening.

> Does this have to be done in a separate preflight? Or could this wait be added right after the daemon start, with a big warning if it fails?

I find it more clear as part of preflight and that's the reason I named it serving so that make sure after start it serve the requests.

Reply:
>  it starts in 7-8 secs but sometime it takes around 10-11 secs when I have lot of I/O happening.

What is taking so long before the http endpoint is available? This definitely would be useful data to provide in the commit log.

Reply:
> @cfergeau I think till now checking crc daemon process is always worked with running functionality because it is started by some other process (tray) or manually , in both situation this other process took the burden of serving (in case of tray we already waited 8-10 sec) and in case of manual we wait until it start serving.

All I mean, is that (for example) the helper introduced in #3151 is named `daemonRunning`, not `daemonServing`, and if you look at other code doing similar checks, it's always using `running` to describe the check being done (`olderDaemonRunning` for example). We never used `serving` before, and I don't think we need the distinction between `running` and `serving`. So introducing `serving` in this PR is just making things harder to understand.

Reply:
> I am intend to look into launchd stuff if even after launchd load the process take some time to even start with launchd but it might take some more time and may be for upcoming release.

Are you saying that these 7/8 seconds are caused by launchd?

Reply:
@cfergeau I gave it a bit thought and right now I think going with serving make sense (for release/blocker) and after that we can put needed effort to pin point this issue (why it takes more than 3-5 secs, is it because of launchd or something else) . Also we can improve around #3151 helper to use it with this preflight. But as of now I just want to go with the workable solution even if is not best. I am happy to take PR if you put something fast what you suggested and also test it. Tomorrow morning I am going to build final bits for release.

Reply:
No going with serving does not make sense as all the other recent PRs have been using `running` while doing almost the same tests. And since this adds an additional preflight, this adds 'serving' in user-visible places, so harder to change later. 
To be frank, for something last minute, this PR is quite big (additional preflight, changes in the unit tests, ...). If you want something minimal which we can refine later, then adding a small retry loop right after starting the daemon with launchd/as a scheduled task is a much smaller PR (and thus easier to review, hopefully less risky, ...)

Reply:
@cfergeau removed the user facing preflight check.